products/billingbudget/api.yaml

```diff
@@ -172,16 +172,22 @@ objects:
             properties:
               - !ruby/object:Api::Type::String
                 name: pubsubTopic
-                required: true
```

Comment:
What's the reasoning behind removing this?

Reply:
When the `monitoringNotificationChannels` field was added `pubsubTopic` was changed from "required" to "optional", [see details for the `allUpdatesRule` object in the API reference](https://cloud.google.com/billing/docs/reference/budget/rest/v1beta1/billingAccounts.budgets#allupdatesrule).

Reply:
Makes sense! Since we're removing this, we'll still need some sort of requirement to ensure we aren't setting up an empty block.  We can achieve this by putting:
```
                at_least_one_of:
                  - budget.0.all_updates_rule.0.pubsub_topic
                  - budget.0.all_updates_rule.0.monitoring_notification_channels
```
in both `pubsubTopic` and `monitoringNotificationChannels`. See https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/monitoring/api.yaml#L724 as an example!

Reply:
Done! Thanks for the example, have added these requirements